examples/api/session.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import os
+
+from restfly.session import APISession
+
+from .users import UsersAPI
+
+
+class GithubAPI(APISession):
+    """
+    Github API wrapper
+
+    Docs:
+        https://developer.github.com/v3/
+
+    Args:
+        url: The github API url
+        token: The user github token
+    """
+
+    _log = logging.getLogger('GithubLogger')
+
+    def __init__(self, **kwargs):
```

Comment:
For something as simple as this, I think a better pattern would be like this one instead:

```suggestion
    def __init__(self, 
        url: str = 'https://api.github.com',
        token: str | None = None,
        **kwargs
    ):
        if not token:
            token = os.getenv('GITHUB_TOKEN')
        if not token:
            raise ConnectionError('No Github token has been provided')
```

Reply:
updated as per your suggestion with little tweaks.